### PR TITLE
EOS-24425: config_apply - Added unittests.

### PR DESCRIPTION
**Pull request by shriya-deshmukh**
# Problem Statement
- EOS-24425: config_apply
- Added unittest. 
- Fixed test_setup.py issue. 
- Removed lyve-cloud references from config.yaml files.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: test/setup/test_provisioner.py [hunk 1]
```diff
@@ -53,5 +54,87 @@ def test_config_apply_bootstrap(self):
...
+
+        cluster_conf_file_path = "/tmp/invalid_cluster.yaml"
+        data = {
+            "cluster": {
+                "name": "cortx-cluster",
+                "id": 'C01',
+                "node_types": {},
+                "storage_sets": [
+                    {"name": "storage-set-1"},
+                    {"name": "storage-set-2"}
```
Comment (cdeshmukh) [inline test/setup/test_provisioner.py:121]: why multiple storage set config apply should fail ? I think Motr mini provisioner can check for these storage set configs, on whats supported by it and what not. @ujjwalpl  ?

File: test/setup/test_setup.py [hunk 2]
```diff
@@ -44,22 +46,17 @@ def test_config_apply(self):
...
-#            argv = [ 'cluster', 'bootstrap', '1', cortx_conf_url ]
-#            cmd = Cmd.get_command(sys.modules['cortx.setup.cortx_setup'], 'test_setup', argv)
-#            self.assertEqual(cmd.process(), 0)
-#
-#        except Exception as e:
-#            print('Exception: ', e)
-#            sys.stderr.write("%s\n" % traceback.format_exc())
-#            rc = 1
-#        self.assertEqual(rc, 0)
+        rc = 0
```
Comment (cdeshmukh) [inline test/setup/test_setup.py:49]: what code changed exactly here?

File: test/setup/test_provisioner.py (hunk 1 above)
Comment (shriya-deshmukh) [reply test/setup/test_provisioner.py:121]: >>why multiple storage set config apply should fail ?
It should not fail, need some modification in test case, will update.

File: test/setup/test_setup.py [hunk 3]
```diff
@@ -44,22 +46,17 @@ def test_config_apply(self):
...
+        try:
+            argv = ['cluster', 'bootstrap', '1', cortx_conf_url]
```
[the remark this answers is not in the document]
Comment (shriya-deshmukh) [reply test/setup/test_setup.py:51]: node_id. 
This we need to remove once the selva's PR is merged, he is going to add code in cluster_bootstrap for reading this value directly from /etc/machine_id

File: test/setup/test_provisioner.py (hunk 1 above)
Comment (cdeshmukh) [reply test/setup/test_provisioner.py:121]: Ok

**Review by ujjwalpl — APPROVED** [review]

File: test/setup/test_setup.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cdeshmukh) [reply test/setup/test_setup.py:51]: @selvakumaar5496's PR should change this code then when available

File: test/setup/test_setup.py (hunk 2 above)
Comment (cdeshmukh) [reply test/setup/test_setup.py:49]: oh rt, i missed its not part of separate test /  function

**Review by ujjwalpl — APPROVED** [review]

File: test/setup/test_multiple_storage_sets.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+cortx_conf_url = "yaml:///tmp/test_conf_store.conf"
+
+
+class TestMultipleStorageSet(unittest.TestCase):
+    """ Test multiple storage_sets """
+
+    def test_for_multiple_storage_sets(self):
+        """ Test for multiple storage sets """
+
+        cluster_conf_file_path = "/tmp/invalid_cluster.yaml"
```
Comment (cdeshmukh) [inline test/setup/test_multiple_storage_sets.py:33]: so what is invalid in this config ?

**Review by cdeshmukh — APPROVED** [review]

**Review by ujjwalpl — APPROVED** [review]

File: test/setup/test_multiple_storage_sets.py (hunk 4 above)
Comment (shriya-deshmukh) [reply test/setup/test_multiple_storage_sets.py:33]: No there is nothing invalid in config for this test, it has multiple storage sets.
Changed file name to cluster.yaml.

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**